File: lab1/Ring.cpp
#include "Ring.h"

Ring::Ring(int r1, int r2) : Shape(0, 0), r1(r1), r2(r2) {}

Ring::Ring(int x, int y, int r1, int r2) : Shape(x, y), r1(r1), r2(r2) {}

int Ring::left() {
	return x;
}

int Ring::right() {
	return x + (r1 > r2 ? r1 : r2) * 2;
}

int Ring::top() {
	return y;
}

int Ring::bottom() {
	return y + (r1 > r2 ? r1: r2) * 2;
}

int Ring::radius2() {
	return (r1 > r2 ? r1 : r2) * (r1 > r2 ? r1 : r2);
}

int Ring::diameter() {
	return (r1 > r2 ? r1 : r2) * 2;
}

File: lab2/Models/Ring.cs
namespace ShapesDatabase.Models;

public class Ring : Shape {
    public int R1 { get; set; }
    public int R2 { get; set; }

    public override int left() => X - R2;
    public override int right() => X + R2;
    public override int top() => Y - R2;
    public override int bottom() => Y + R2;

    public int radius2() => (R1 > R2 ? R1 : R2) * (R1 > R2 ? R1 : R2);
    public int diameter() => (R1 > R2 ? R1 : R2) * 2;

    public Ring(int r1, int r2) : base(0, 0) {
        R1 = r1;
        R2 = r2;
    }

    public Ring(int x, int y, int r1, int r2) : base(x, y) {
        R1 = r1;
        R2 = r2;
    }

    public Ring(int x, int y, int r1, int r2, int asdasdas) : base(x, y) {
        R1 = r1;
        R2 = r2;
    }
}